libs/util/shared/src/lib/services/theme.service.ts

```diff
@@ -10,6 +10,22 @@ export class ThemeService {
     return document.documentElement.style.getPropertyValue(`--color-${name}`)
   }
 
+  static generateBgOpacityClasses(
+    colorName,
+    colorValue,
+    opacities = [0, 25, 50, 75]
+  ) {
+    const color = chroma(colorValue)
+    const styleElement = document.createElement('style')
+    styleElement.innerHTML = opacities.reduce((cssRules, opacity) => {
+      cssRules += `.bg-${colorName}-opacity-${opacity}{background-color:${color
+        .alpha(opacity / 100)
+        .css()};}`
+      return cssRules
+    }, '')
+    document.getElementsByTagName('head')[0].appendChild(styleElement)
+  }
```

Comment:
As discussed before, I'm really not a big fan of this solution. To work properly the buttons in the UI will rely on custom CSS classes generated by code, which is inconsistent with the rest of the styling workflow (based on Tailwind). Besides, searching for "gb-primary-opacity-25" for example will yield 0 results, so this will typically be the kind of thing that newcomers will have a hard time understanding.

The alternative is to have a separate element for the button background, and have its opacity be changed without interfering with the button content, which is IMO not a hack but a common situation when writing CSS/HTML. Besides, this can be handled in the `gn-ui-button` component to avoid adding verbosity in the app.

Letting both authors judge on this :)

Reply:
I've digged a bit on how TW handles this, and actually, it can't be covered by TW.
TW actually does not use css-variables for the colors, which are hardcoded in the styles.
For the opacity, here is the definition:
```css
#bg-blue-500 
--tw-bg-opacity: 1; 
background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
```
Then you can override `--tw-bg-opacity` in your component to have the correct opacity
```html
<div class="bg-purple-600 bg-opacity-50 ..."></div>
```

This would work if we declared in tw.config.css something like
```js
primary: `#435465`
```
but because we are refering to a css variables here (which is not really how TW is meant to work IMO), 
```js
primary: 'var(--color-primary)',
```
then the process of managing the opacity with `var(--tw-bg-opacity)` and `var(--tw-text-opacity)` does not work, which is a bit weird some how. I have declared the `--color-primary` as `rgba(59, 130, 246, var(--tw-bg-opacity)` but it seems that TW, when creating the `bg-primary` class, affects the value which is computed at the root of the html, where the opacity is `1`, whatever you change the `--tw-bg-opacity` underneath, it's not evaluated and keeps being `1`, which I don't really understand.

That said, the correct solution would be to manually add some css classes, like `bg-primary-opacity-25` in styles.css, the problem is that, because it's a css variables, we can't compute the color value in alpha.

If you create a component for that, anyway you will have to compute the opacity color in javascript, and you won't have any class for this either, it could be ok if it's only used on button but those variables could be used in other components, like panel header background, badges on so on, so you would duplicate all the code in all component and create many components just to handle a background opacity, which is not a good usecase IMO.

Maybe we could add a documentation, or add empty css rules of all the rules we create manually, as an extension of TW preset ?

Reply:
> If you create a component for that, anyway you will have to compute the opacity color in javascript, and you won't have any class for this either, it could be ok if it's only used on button but those variables could be used in other components, like panel header background, badges on so on, so you would duplicate all the code in all component and create many components just to handle a background opacity, which is not a good usecase IMO.

Or the gn-ui-button component could be customizable using a variable such as `--gn-ui-button-background-opacity-inactive`. Having a button with a transparent background is a pretty common usecase IMO, so it might make sense to offer this possibility.

Reply:
`--gn-ui-button-background-opacity-inactive` is a component input or a css var i don't get it ?
Anyway, we have button, but also a badge, and maybe tomorrow a panel header or an icon ...

A question, you are creating css variables in `ThemeService`, and, the same way, the user is not aware about the fact that they are created and that they exist right ? Exactly the same as the new css rules ?

A solution could be to do a directive... but to me, managing this specific use case is a bit too complex just because we can't hard code a css rule in a css file.

Reply:
> `--gn-ui-button-background-opacity-inactive` is a component input or a css var i don't get it ?

A CSS variable that the gn-ui-button component would use in its stylesheet, with a default fallback value. Then an app could give a value to this variable and that would change the appearance of one or several buttons.

> A question, you are creating css variables in ThemeService, and, the same way, the user is not aware about the fact that they are created and that they exist right ? Exactly the same as the new css rules ?

Hmm, the variables created this way are used by Tailwind classes, so the user should not have to think about them.

Anyway, it's probably not necessary to go too far just for two buttons, so I guess we can keep this as it is for now. @tkohr what do you think?

Reply:
Hum, I see your respective points (more or less), but would have a hard time to judge on this. That's why I haven't added my two more cents to it;-) But as you say, I'd also propose to keep it as is for now, regarding the limited scope.